rsciio/digitalsurf/_api.py

```diff
@@ -45,12 +46,82 @@
 # import rsciio.utils.tools
 # DictionaryTreeBrowser class handles the fancy metadata dictionnaries
 # from hyperspy.misc.utils import DictionaryTreeBrowser
-from rsciio._docstrings import FILENAME_DOC, LAZY_UNSUPPORTED_DOC, RETURNS_DOC
+from rsciio._docstrings import (
+    FILENAME_DOC,
+    LAZY_UNSUPPORTED_DOC,
+    RETURNS_DOC,
+    SIGNAL_DOC,
+)
+from rsciio.utils.date_time_tools import get_date_time_from_metadata
 from rsciio.utils.exceptions import MountainsMapFileError
+from rsciio.utils.rgb_tools import is_rgb, is_rgba
 
 _logger = logging.getLogger(__name__)
 
 
+def parse_metadata(cmt: str, prefix: str = '$', delimiter: str = '=') -> dict:
+    """Parse metadata from the comment field of a digitalsurf file, or any other
+    str in similar formatting. Return it as a hyperspy-compatible nested dict.
+
+    Parameters
+    ----------
+    cmt: str
+        str containing contents of a digitalsurf file comment field
+    prefix: str, default = '$' 
+        prefix character that must start each line, otherwise line is ignored.
+        '$' for digitalsurf files, typically '' when parsing from text files
+    delimiter: string, default = '='
+        Character that delimit key-value pairs in digitalsurf comments. 
+        Usually '='
+
+    Returns
+    -------
+    dict_md: nested dictionnary.
+
+    """
```

Comment:
Some docstring formatting fixes:

```suggestion
def parse_metadata(cmt: str, prefix: str = "$", delimiter: str = "=") -> dict:
    """
    Parse metadata from the comment field of a digitalsurf file, or any other
    str in similar formatting. Return it as a hyperspy-compatible nested dict.

    Parameters
    ----------
    cmt : str
        String containing the contents of a digitalsurf file comment field.
    prefix : str
        Prefix character that must start each line, otherwise line is ignored.
        ``"$"`` for digitalsurf files, typically an empty string (``""``) when
        parsing from text files.
        Default is ``"$"``.
    delimiter : str
        Character that delimit key-value pairs in digitalsurf comments.
        Default is ``"="``.

    Returns
    -------
    dict
        Nested dictionnary of the metadata.

    """
```